### PR TITLE
V2 deprecation removals

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Closes #{ID}

## Description

This is a PR of all the deprecations for V2, aim to keep it up to date with develop and any warnings removed

## Interface Changes

- ComponentManager can no longer be imported from `base/builder.py` import from `base/reactor.py`

`reactor.py`
- `dim` is no longer an allowed kwarg for show_cad use `dims` 
- `filter_` -> `component_filter`
- `formatt` -> `cad_format`

`bluemira/builders/tf_coils.py`
- `ripple_selector` must be specified

`bluemira/codes/_freecadapi.py`
- save_as_STP `scale` -> `unit_scale` as a string

`bluemira/utilities/plot_tools.py`
- `make_gif` and `save_figure`: `formatt` -> `file_format`

`bluemira/geometry/tools.py`
- `save_cad`: `formatt` -> `cad_format`

`bluemira/equilibria/fem_fixed_boundary/file.py`
- `save_fixed_boundary_to_file`: `formatt` -> `file_format`

### `equilibria`
- Removed files:
    opt_constraints.py
    opt_constraints_funcs.py
    opt_objectives.py
    opt_problems.py
    positioner.py
- Replaced by the `equilibria/optimisation` module
- positioner.py has been superceeded by utilities/positioning.py

### `geometry`
- Removed functions -`GeometryOptimisationProblem`, `minimise_length`, `MinimiseLengthGOP`
- Replaced with GeomOptimisationProblem` and the objective function 
  ```python
  def f_objective(self, geom) -> float:
       return geom.create_shape().length
   ```   
### `utilities` 
most of these base classes have been moved to the optimisation module.
- Removed files:
  opt_problems.py
  opt_tools.py
  optimiser

`builders/coil_supports::StraightOISOptimisationProblem` no longer requires the optimiser argument, the `StraightOISDesigner` could in future add the customisations to the optimiser if required.

`geometry.paramerisations` no longer have the function `shape_ineq_constraints` please use the `f_ineq_constraint` and optionally the `df_ineq_constraint` when creating your own paramterisation optimisations otherwise the conversion has already occured as part of the optimisation refactor.

## Other changes

- Boolean traps no longer exist and typing with Union, List etc has been replaced with their builtin types 
- British english enforced in user facing APIs

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by je-cook** [comment]
To go in after #2887

**Review by oliverfunk — APPROVED** [review]
so great! LGTM! 👏🏼

File: bluemira/codes/_nlopt_api.py
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/codes/_nlopt_api.py:1]: here: 

https://github.com/Fusion-Power-Plant-Framework/bluemira/tree/develop/bluemira/optimisation/_nlopt

**Comment by je-cook** [comment]
@CoronelBuendia could you do a review for the small changes in radiation and plasma physics? It's all boolean trap changes IIRC:
https://docs.astral.sh/ruff/rules/boolean-default-value-positional-argument/

**Review by CoronelBuendia — APPROVED** [review]

**Pull request merged**